source/slang/slang-ir-autodiff-transcriber-base.cpp

```diff
@@ -1075,4 +1121,64 @@ InstPair AutoDiffTranscriberBase::transcribeInst(IRBuilder* builder, IRInst* ori
     return result;
 }
 
+
+void AutoDiffTranscriberBase::markDiffTypeInst(IRBuilder* builder, IRInst* diffInst, IRType* primalType)
```

Comment:
It marks insts of `T : IDifferentiable` as differential (to move into the derivative function & transpose) and `T : IDifferentiablePtrPair` as primal (to move it to the primal blocks, and not consider it during transposition)

Reply:
line 1147 is what I don't understand.
if `diffInst` isn't an `IRType`, then it can only be one of lookupWitness, specialization, extractExistentialType. How is this related to the subject of this function?

Reply:
Ah, I think the misunderstanding may be due to what `diffInst` is supposed to be. 
Here `diffInst` is any inst (not just a type) that is produced during forward-mode as the derivative counter-part for the primal inst. It is not the type of the inst, but the inst itself, which can be an `IRType` for cases where we construct types within the method body). 
This method is responsible for deciding if this inst requires further processing or not by marking it appropriately.

Perhaps `markDiffInst` is a better thing to do here.